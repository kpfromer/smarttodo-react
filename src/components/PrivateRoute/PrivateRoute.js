import React, { PureComponent } from 'react';
import { Redirect, Route } from "react-router-dom";
import { connect } from "react-redux";

export class PrivateRoute extends PureComponent {

  render() {
    const { component: Component, isAuthenticated, ...rest } = this.props;
    return (
      <Route {...rest} render={(props) => (
        isAuthenticated
          ? <Component {...props} />
          : <Redirect to={{
            pathname: '/login',
            // state: { from: props.location }
          }} />
      )} />
    );
  }
}

export const mapStateToProps = state => ({
  isAuthenticated: !!state.login.token
});

export default connect(mapStateToProps)(PrivateRoute);