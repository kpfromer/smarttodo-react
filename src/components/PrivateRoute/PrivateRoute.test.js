import React from "react";
import { SetupComponent } from "react-component-setup";
import { PrivateRoute } from "./PrivateRoute";

const { shallow: setup } = SetupComponent({
  component: PrivateRoute
});

const CustomComponent = () => (
  <h1>hello world</h1>
);

describe('PrivateRoute', () => {
  let component, routerProps;

  beforeEach(() => {
    component = CustomComponent;
    routerProps = {
      hello: 'world',
      my: 'name is kyle'
    };
  });

  it('renders without crashing', () => {
    setup();
  });

  it('renders a route to component if `isAuthenticated`', () => {
    const { wrapper } = setup({
      component,
      isAuthenticated: true
    });

    expect(wrapper.props().render(routerProps)).toMatchSnapshot();
  });

  it('renders a route that redirects users to `/login` if not `isAuthenticated`', () => {
    const { wrapper } = setup({
      component,
      isAuthenticated: false
    });

    expect(wrapper.props().render(routerProps)).toMatchSnapshot();
  });

  it('passes the rest of it\'s props to the Route', () => {
    const { wrapper } = setup({
      component,
      isAuthenticated: false,
      otherProps: 'needed',
      to: '/an-example-location'
    });

    expect(wrapper).toMatchSnapshot();
  });
});