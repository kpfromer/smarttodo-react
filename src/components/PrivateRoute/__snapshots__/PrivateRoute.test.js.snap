// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute passes the rest of it's props to the Route 1`] = `
<Route
  otherProps="needed"
  render={[Function]}
  to="/an-example-location"
/>
`;

exports[`PrivateRoute renders a route that redirects users to \`/login\` if not \`isAuthenticated\` 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/login",
    }
  }
/>
`;

exports[`PrivateRoute renders a route to component if \`isAuthenticated\` 1`] = `
<CustomComponent
  hello="world"
  my="name is kyle"
/>
`;
